### DISCUSSION TITLE
IPAM model to create and delete host.

### DISCUSSION THREAD

**Discussion by john2893** [netbox-community/netbox#14849] [Help Wanted!] [answered]
I'm trying to create a custom script/form to create and delete host.
The form will take - hostname and IPaddress with mask. 

```

from extras.scripts import *
from ipam.models import IPAddress, VRF
from dcim.models import Device

    hostname  = StringVar(
        description="hostname"
    )

    ip-address = IPAddressWithMaskVar(
        description="IP",
    )


def run(self, data, commit):
           new_ip = IPAddress(
            address="192.168.1.10",
            status="ACTIVE",
            dns_name = 'test-do.example'
            description="test-ip",
        )
        new_ip.save()

```
The netbox gui does not accept this as in the it just throws the error in the web:
Script file at /usr/local/netbox/custom/new-form.py could not be loaded.

**Comment by candlerb** [comment]
Your script has a syntax error which prevents it being loaded, but unfortunately Netbox doesn't show you what the actual error is, unless you enable logging (see #12766)

Try adding the following to the very top of your script:

```
import django, os, sys
sys.path.append('/opt/netbox/netbox')
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'netbox.settings')
django.setup()
```

Then run it again using:

```
/opt/netbox/venv/bin/python /usr/local/netbox/custom/new-form.py
```

This ought to be a FAQ.

**Comment by john2893** [comment]
Thank you!
But I was going to run it from netbox web GUI -- > Customization -->
Scripts --> New-Form

This looks like its set to run CLI way?

**Reply by candlerb** [reply, marked as answer]
You have to run it this way to see the syntax error. Once you've fixed the syntax error, delete or comment out those extra lines, then you can run it from the GUI. (The script won't actually *run* from the command line this way anyway. It will just load all the classes, then terminate).

**Reply by john2893** [reply]
Thank you! 
I had a tough time with getting the DJANGO_SETTINGS_MODULE variable set to work. 

But I copied one of the example script and then added my script bit by bit. That got the script loaded.

ip.save()  ---> does the saving. 
ip.delete() --> deletes

I am looking at https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/models/ip.py trying to find the best way to output the log for a successful add and delete from netbox.
